clippy_lints/src/undocumented_unsafe_blocks.rs

```diff
@@ -546,7 +580,7 @@ fn get_body_search_span(cx: &LateContext<'_>) -> Option<Span> {
     for (_, node) in map.parent_iter(body.hir_id) {
         match node {
             Node::Expr(e) => span = e.span,
-            Node::Block(_) | Node::Arm(_) | Node::Stmt(_) | Node::Local(_) => (),
+            Node::Block(_) | Node::Arm(_) | Node::Stmt(_) | Node::Local(_) | Node::Item(_) => (),
```

Comment:
I am not 100% sure that this change would not lead to undesirable side effects. This was necessary to make it work for the const/static cases.
Wdyt?

Reply:
a `Node::Item` is just an item, i.e., `fn`, `const`, `mod`, etc, so while I don't think that would cause any issues you can probably check that `kind` is either `Static` or `Const`

Reply:
Ok! I have done that and ammend to the previous commit. https://github.com/rust-lang/rust-clippy/pull/10886/commits/e9562675ed0990fe54ffd96fbc2237de803bbc74